include/dba.php

```diff
@@ -662,17 +682,35 @@ static public function e($sql) {
 
 		$args = func_get_args();
 
-		$stmt = call_user_func_array('self::p', $args);
+		// In a case of a deadlock we are repeating the query 20 times
+		$timeout = 20;
 
-		if (is_bool($stmt)) {
-			$retval = $stmt;
-		} elseif (is_object($stmt)) {
-			$retval = true;
-		} else {
-			$retval = false;
-		}
+		do {
+			$stmt = call_user_func_array('self::p', $args);
 
-		self::close($stmt);
+			if (is_bool($stmt)) {
+				$retval = $stmt;
+			} elseif (is_object($stmt)) {
+				$retval = true;
+			} else {
+				$retval = false;
+			}
+
+			self::close($stmt);
+
+		} while ((self::$dbo->errorno == 1213) && (--$timeout > 0));
```

Comment:
@tobiasd The error 1213 describes a deadlock. This can happen when a table is heavily used - like the workerqueue table. I experienced the problem very often when I increased the speed. The proposed for a deadlock is to retry the transaction.